packages/flutter/lib/src/painting/binding.dart

```diff
@@ -44,6 +46,11 @@ abstract class PaintingBinding extends BindingBase with ServicesBinding {
   @protected
   ImageCache createImageCache() => ImageCache();
 
+  /// Calls through to [dart:ui] with [decodedCacheRatioCap] from [ImageCache].
+  Future<ui.Codec> instantiateImageCodec(Uint8List list) {
+    return ui.instantiateImageCodec(list, decodedCacheRatioCap: _imageCache.decodedCacheRatioCap);
+  }
```

Comment:
might be good to put this method on ImageCache itself, or put the configuration option on the binding, so that they're together. It's a bit weird to have the configuration so distant from the use site.

Reply:
Moved it to the binding. I think that makes a bit more sense since this isn't really going to `ImageCache`, it's going to `Codec`.

Reply:
SGTM.

FWIW, the reason to put it on another object is that if you override this, you may not care about the cache ratio cap, but since it's on the superclass you can't get rid of it. If it was on its own object (or the image cache), you would replace the lot in one go (assuming that they implemented an interface that didn't include these fields, etc). But it's not important (and doing that would be a lot of faffing around for minimal gain anyway.)

Reply:
OK, makes sense. I think it's probably better to keep this here for now still.